tools/pipeline-generator/Azure.Sdk.Tools.PipelineGenerator/Conventions/PipelineConvention.cs

```diff
@@ -241,19 +242,31 @@ private bool EnsureReportBuildStatus(BuildDefinition definition)
             return hasChanges;
         }
 
-        /// <summary>
-        /// Start hour (3AM)
-        /// </summary>
-        protected const int StartHourOffset = 3;
+        protected const int FirstSchedulingHour = 0;
+        protected const int LastSchedulingHour = 12;
+        protected const int BucketSizeInMinutes = 15;
 
-        /// <summary>
-        /// Number of buckets for hour hashing
-        /// </summary>
-        protected const int HourBuckets = 3;
-
-        protected int HashBucket(string pipelineName)
+        protected Schedule CreateScheduleFromDefinition(BuildDefinition definition)
         {
-            return pipelineName.GetHashCode() % HourBuckets;
+            var totalHours = LastSchedulingHour - FirstSchedulingHour;
+            var totalMinutes = totalHours * 60;
+            var totalBuckets = totalMinutes / BucketSizeInMinutes;
+
+            var bucket = definition.Id % totalBuckets;
+            var startHours = bucket / 4;
```

Comment:
Shouldn't 4 be (60 / BucketSizeInMinutes) so it is still a function of the constants?